Single print for Deprecated Cell.add_surface(...) method

wbinventor opened the pull request:
The OpenCG-based BEAVRS builder in mit-crpg/PWR_benchmarks throws a large number of the following `DeprecationWarnings`:

```bash
/usr/local/lib/python2.7/dist-packages/openmc-0.7.0-py2.7.egg/openmc/universe.py:247: DeprecationWarning: Cell.add_surface(...) has been deprecated and may be removed in a future version. The region for a Cell should be defined using the region property directly.
```

These warnings were introduced in PR #463 with complex regions. OpenCG does not (yet) support complex regions, and hence the OpenMC-OpenCG compatibility module still uses the `Cell.add_surface(...)` method. These `DeprecationWarnings` really cloud the output when one runs the BEAVRS builder which can annoying and distracting, esp. to those users who have no reason to know the reason for this. 

This PR makes it such that this warning is only printed to the console once. Perhaps this is not good practice since it is less transparent than before - if so, we can at least discuss the best course of action here. Of course the "real fix" is to implement complex regions in OpenCG (which could likely be based largely upon the current implementation in the OpenMC Python API). But this doesn't seem to be a priority for anyone at the moment, hence the need for this PR to bridge the gap.

paulromano commented:
Why not just filter the warnings in the OpenCG builder itself?

wbinventor commented:
I tried putting a filter on the warnings in the BEAVRS builder - both
before and after importing OpenMC - and it didn't work.
On Dec 7, 2015 7:50 AM, "Paul Romano" <notifications@github.com> wrote:

> Why not just filter the warnings in the OpenCG builder itself?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-crpg/openmc/pull/521#issuecomment-162516396>.
>

paulromano commented:
The problem is that
```
warnings.simplefilter('always', DeprecationWarning)
```
is called inside of `add_surface(...)` thereby overriding any filter set elsewhere. We should just move that to the top of the universe module so that it only gets called once when the module is imported. Then, as long as you filter warnings in the builder after openmc has been imported, you'll get the warning only once. Capiche?

wbinventor commented:
Thanks for the advice @paulromano - I've moved the `DeprecationWarning` filter to the top of the "universe.py" module, and reverted to "always" reporting the warnings. I'll submit a tiny PR to mit-crpg/PWR_benchmarks with the appropriate warning filter shortly.
